src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h

```diff
@@ -149,18 +173,8 @@ class STSbyGroup : public Variable::IVariable<STSbyGroup<NextT>, NextT, VCardSTS
         pValuesForTheCurrentYear = new VCardType::IntermediateValuesBaseType[pNbYearsParallel];
 
         // Building the vector of group names the clusters belong to.
-        std::set<std::string> tmp_names;
-        for (const auto& cluster : area->shortTermStorage.storagesByIndex)
-            tmp_names.insert(cluster.properties.groupName);
-        groupNames_ = {tmp_names.begin(), tmp_names.end()};
-
-        // Giving a number to each group
-        unsigned int groupNumber{0};
-        for (const auto name : groupNames_)
-        {
-            groupToNumbers_[name] = groupNumber;
-            groupNumber++;
-        }
+        groupNames_ = sortedUniqueGroups(area->shortTermStorage.storagesByIndex);
+        groupToNumbers_ = addNumbersToGroups(groupNames_);
```

Comment:
**addNumbersToGroups**  ==>  **giveNumbersToGroups**, more accurate it seems

Reply:
OK